### PR TITLE
Upgrade script

### PR DESCRIPTION
**Pull request by bhale**
Usage: 

K3OS_VERSION="v0.4.0" /sbin/k3os-upgrade

With much debt to @zimme

### CODE REVIEW

File: overlay/sbin/k3os-upgrade [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	ARCH="arm"
+else
+	echo "Unsupported CPU architecture."
+	exit 0
+fi
+
+echo "Upgrading k3os to ${VERSION}"
+
+cd /k3os/system/
+mount -o remount rw .
```
Comment (dweomer) [inline overlay/sbin/k3os-upgrade]: Given the remount (and the permissioned untar/copy below) please implement a check at the top of this script to make sure we are running as `root` and then nix any usage of `sudo`.

File: overlay/sbin/k3os-upgrade [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+	ARCH="arm64"
+elif [ $PROC == "armv7l" ]
+then
+	ARCH="arm"
+else
+	echo "Unsupported CPU architecture."
+	exit 0
+fi
+
+echo "Upgrading k3os to ${VERSION}"
```
Comment (dweomer) [inline overlay/sbin/k3os-upgrade]: should be
```
echo "Upgrading k3os to ${K3OS_VERSION}"
```
?

File: overlay/sbin/k3os-upgrade [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+cd /k3os/system/
+mount -o remount rw .
+curl -LO https://github.com/rancher/k3os/releases/download/${K3OS_VERSION}/k3os-rootfs-${ARCH}.tar.gz
+tar xfz k3os-rootfs-${ARCH}.tar.gz; sudo cp -R ${K3OS_VERSION}/k3os/system/* .
+rm -rf k3os-rootfs-${ARCH}.tar.gz ${K3OS_VERSION}
```
Comment (dweomer) [inline overlay/sbin/k3os-upgrade]: Maybe something like this is cleaner:
```
K3OS_ROOTFS_URL="https://github.com/rancher/k3os/releases/download/${K3OS_VERSION}/k3os-rootfs-${ARCH}.tar.gz"
curl -fsSL "${K3OS_ROOTFS_URL}" | tar xz --strip-components=3
```

Comment (dweomer) [reply overlay/sbin/k3os-upgrade]: Would also prefer this be renamed to `k3os-upgrade-rootfs`

Comment (zimme) [reply overlay/sbin/k3os-upgrade]: The current code won't work actually as when you download from "latest" the extracted files will be under a specific version number folder and not a folder named "latest".

Comment (bhale) [reply overlay/sbin/k3os-upgrade]: Cool, adopted these changes.

File: overlay/sbin/k3os-upgrade (hunk 2 above)
Comment (bhale) [reply overlay/sbin/k3os-upgrade]: Yup.

File: overlay/sbin/k3os-upgrade (hunk 1 above)
Comment (bhale) [reply overlay/sbin/k3os-upgrade]: Done.

**Comment by zimme** [comment]
This is no longer needed as the tar command extracts the files to the right place directly. If I understood the change to the tar command correctly.
```
cp -R ${K3OS_VERSION}/k3os/system/* .
rm -rf k3os-rootfs-${ARCH}.tar.gz ${K3OS_VERSION}
```

**Pull request merged**